{"additions": 1, "auther_ref": "bugFixLumiStatus", "auther_sha": "2319d1552b0d61d13c426908fd457bc72f987d2c", "author": "wddgit", "body": "#### PR description:\r\n\r\nWe are having a rare, nonreproducible failure in\r\nEventProcessor::readLuminosityBlock in the IBs.\r\nThe symptom is an assert because the pointer to\r\nthe LuminosityBlockPrincipal is null. If the\r\nLuminosityBlockProcessingStatus object was being\r\ndestroyed before calling resumeGlobalLumiQueue\r\nthen such a failure could occur because the order the\r\nof the member variables in LuminosityBlockProcessingStatus\r\nis wrong. We should free the LuminosityBlockPrincipal\r\nbefore resuming the lumi queue. I am not sure this will\r\nactually help though, because I can see no way to get\r\ninto this state. The existing code tries to always call\r\nresumeGlobalLumiQueue, but this should eliminate at least\r\none possible source for the assert we have seen.\r\n\r\nThis should not change any behavior except maybe in some\r\ncomplex case where we are already dealing with an unrelated\r\nexception.\r\n\r\n#### PR validation:\r\n\r\nRun existing unit tests.", "branch": "master", "changed_files": 1, "closed_at": "1560276676", "comments": 11, "commits": 1, "created_at": "1560185095", "deletions": 1, "labels": ["code-checks-approved", "comparison-available", "core-approved", "fully-signed", "orp-approved", "tests-approved"], "merge_commit_sha": "1cd9244e4130c1e196b7c2f4c40164fac52f403d", "merged_at": "1560276676", "merged_by": "cmsbuild", "milestone": "CMSSW_11_0_X", "number": 27158, "release-notes": [], "review_comments": 0, "state": "closed", "title": "Fix assert failure in readLuminosityBlock", "updated_at": "1560276676", "user": "wddgit"}